packages/tre/src/plugins/r2/validator.ts

```diff
@@ -96,17 +101,53 @@ export class Validator {
     return checksums;
   }
 
-  condition(meta?: R2Object, onlyIf?: R2Conditional): Validator {
+  condition(
+    meta?: R2Object | Pick<R2Object, "etag" | "uploaded">,
+    onlyIf?: R2Conditional
+  ): Validator {
     if (onlyIf !== undefined && !_testR2Conditional(onlyIf, meta)) {
       let error = new PreconditionFailed();
-      if (meta !== undefined) error = error.attach(meta);
+      if (meta instanceof R2Object) error = error.attach(meta);
       throw error;
     }
     return this;
   }
 
-  size(value: Uint8Array): Validator {
-    if (value.byteLength > MAX_VALUE_SIZE) {
+  range(
+    options: Pick<R2GetOptions, "rangeHeader" | "range">,
+    size: number
+  ): InclusiveRange | undefined {
+    if (options.rangeHeader !== undefined) {
+      const ranges = _parseRanges(options.rangeHeader, size);
+      // If the header contained a single range, use it. Otherwise, if the
+      // header was invalid, or contained multiple ranges, just return the full
+      // response.
+      if (ranges?.length === 1) {
+        const [start, end] = ranges[0];
+        return { start, end };
+      }
+    } else if (options.range !== undefined) {
+      let { offset, length, suffix } = options.range;
+      // Eliminate suffix is specified
```

Comment:
```suggestion
    // Eliminate suffix if specified
```

Reply:
What does eliminate mean here?
Doesn't the logic here mean that if a suffix length is specified you return only the suffix?